src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs

```diff
@@ -216,6 +216,32 @@ public virtual PartialViewResult PartialView(string viewName, object model)
         /// The view component name. Can be a view component
         /// <see cref="ViewComponents.ViewComponentDescriptor.ShortName"/> or
         /// <see cref="ViewComponents.ViewComponentDescriptor.FullName"/>.</param>
+        /// <returns>The created <see cref="ViewComponentResult"/> object for the response.</returns>
+        [NonAction]
+        public virtual ViewComponentResult ViewComponent(string componentName)
```

Comment:
Instead of these extra methods we could also have default params in the other methods but we generally don't do that. Let me know if you feel differently.